### PR TITLE
Bootstrapify Calendar, including bootstrap subnav

### PR DESCRIPTION
**Pull request by seanclifford**
Implemented as part of issue #958 to standardize the subnav component used on the site.

Style is mostly the same as before with bootstrap grid classes used.
Resizes reasonably nicely now, and the year header and footer dropdown are always centered too.
Additionally moved the custom page CSS to a an `scss` file.

Note: As previously discussed, I made separate commits for the file renames to `haml`, but I believe history of the renamed files will not be "tracked" if these commits are squashed on merge.

### CODE REVIEW

File: app/assets/stylesheets/calendars/index.scss [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  display:inline-table;
+
+  caption {
+    text-align: center;
+  }
+  th {
+    text-align: center;
+  }
+  td {
+    font-size:12px;
```
Comment (kueda) [inline app/assets/stylesheets/calendars/index.scss:31]: Can we get

1. A space after `:`
2. `line-height: 30px` so it's not quite so vertically smushed

File: app/views/calendars/index.html.haml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+        = render :partial => 'calendar', :locals => { :observations => @observations_by_month[i+1], :year => @year, :month => i+1, :month_name => t("date_format.month.#{month}") }
+
+  .row.yearnav
+    .col-xs-4
+      %h2.left
+        = link_to "&lsaquo; #{@year - 1}".html_safe, :year => @year - 1
+    .col-xs-4
+      %h4.center
+        %form( method="get" )
+          = select_tag :year, options_for_select(1950..Date.today.year, :selected => @year), "data-autosubmit" => true
```
Comment (kueda) [inline app/views/calendars/index.html.haml:37]: I think you'll get a slightly more bootstrap-y form with something like

```haml
%form{ method: "get", class: "form-inline" }
  = select_tag :year, options_for_select( 1950..Date.today.year, selected: @year ), "data-autosubmit" => true, class: "form-control"
```

**Review by kueda — CHANGES_REQUESTED** [review]
Mostly looking good! Not sure what to do about squashing vs history. I like both, but maybe history is more important.

File: app/assets/stylesheets/calendars/index.scss (hunk 1 above)
Comment (seanclifford) [reply app/assets/stylesheets/calendars/index.scss:31]: Ah, that looks much better, thanks! 
After changing the line height I noticed the columns were also horizontally smushed around the Thu,Fri,Sat columns. I've made them even widths by adding `table-layout: fixed;`

File: app/views/calendars/index.html.haml (hunk 2 above)
Comment (seanclifford) [reply app/views/calendars/index.html.haml:37]: Thanks, that looks better.
In case you haven't noticed, I'm working out bootstrap as I go along!

**Comment by seanclifford** [comment]
On squashing vs history - I'm personally not too bothered either way... 
Another option would be that I could reauthor the commits down to 2 commits - one for renames, and the rest with the changes on a new branch and PR... But that is a bit of a faff.

**Review by kueda — APPROVED** [review]

**Pull request merged**